10-JavaScript/hoisting.md

```diff
@@ -73,6 +75,103 @@ In the example above, the `foo` function _will not_ be hoisted because it is an
 
 Function `bar` _will_ be hoisted because it is a **declaration**.
 
+## Block-Level Scoping with `let`
+
+Variables, including functions, declared using `let` differ from `var` in that they are scoped to the block-level rather than the function.  
+
+If you use var to declare variables:
+
+```javascript
+function foo() {
+  if (false) {
+    var x = 1;
+  }
+  return;
+  var y = 1;
+}
+```
+
+The JavaScript interpreter will move the declarations to the top of the function as below.  
+
+```javascript
+function foo() {
+  var x,y;
+  if (false) {
+    x = 1;
+  }
+  return;
+  y = 1;
+}
+```
+
+However if we had used `let`
+
+```javascript
+function foo() {
+  if (false) {
+    let x = 1;
+  }
+  return;
+  let y = 1;
+}
+```
+
+The interpreter would look not hoist, the variable declaration to the top of the function.  They would only exist in the scope in which they are declared.
```

Comment:
I think this is still confusing. Maybe "The interpreter will not hoist the variable declaration to the top of the function when using `let`. The interpreter will keep the scope in which these variables are declared."